internal/handlers/pg/msg_findandmodify.go

```diff
@@ -179,8 +208,12 @@ func (h *Handler) MsgFindAndModify(ctx context.Context, msg *wire.OpMsg) (*wire.
 			return &reply, nil
 		}
 
-		_, err = h.delete(ctx, params.sqlParam, resDocs)
+		_, err = h.delete(runCtx, params.sqlParam, resDocs)
 		if err != nil {
+			if errors.Is(err, context.DeadlineExceeded) {
+				return nil, fmt.Errorf("context deadline exceeded, maxTimeMS: %v", params.maxTimeMS)
```

Comment:
Why do we need that? Why can't we return errors as they are?

Reply:
We can, I removed it